apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { VirtualList } from '@/components';
+import { useCurrentAccount, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+import { useMigrationTransaction } from '@/hooks/useMigrationTransaction';
+import {
+    Button,
+    Dialog,
+    Header,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    KeyValueInfo,
+    LoadingIndicator,
+    Panel,
+    Title,
+    TitleSize,
+} from '@iota/apps-ui-kit';
+import { useGroupedMigrationObjectsByExpirationDate } from '@/hooks';
+import { Loader, Warning } from '@iota/ui-icons';
+import { DialogLayout, DialogLayoutBody, DialogLayoutFooter } from './layout';
+import { MigrationObjectDetailsCard } from '../migration/migration-object-details-card';
+import { Collapsible, useFormatCoin } from '@iota/core';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { summarizeMigratableObjectValues } from '@/lib/utils';
+import toast from 'react-hot-toast';
+
+interface MigrationDialogProps {
+    basicOutputObjects: IotaObjectData[] | undefined;
+    nftOutputObjects: IotaObjectData[] | undefined;
+    onSuccess: (digest: string) => void;
+    setOpen: (bool: boolean) => void;
+    open: boolean;
+    isTimelocked: boolean;
+}
+
+export function MigrationDialog({
+    basicOutputObjects = [],
+    nftOutputObjects = [],
+    onSuccess,
+    open,
+    setOpen,
+    isTimelocked,
+}: MigrationDialogProps): JSX.Element {
+    const account = useCurrentAccount();
+    const {
+        data: migrateData,
+        isPending: isMigrationPending,
+        isError: isMigrationError,
+    } = useMigrationTransaction(account?.address || '', basicOutputObjects, nftOutputObjects);
+
+    const {
+        data: resolvedObjects = [],
+        isLoading,
+        error: isGroupedMigrationError,
+    } = useGroupedMigrationObjectsByExpirationDate(
+        [...basicOutputObjects, ...nftOutputObjects],
+        isTimelocked,
+    );
+
+    const { mutateAsync: signAndExecuteTransaction, isPending: isSendingTransaction } =
+        useSignAndExecuteTransaction();
+    const { totalNonOwnedStorageDepositReturnAmount } = summarizeMigratableObjectValues({
+        basicOutputs: basicOutputObjects,
+        nftOutputs: nftOutputObjects,
+        address: account?.address || '',
+    });
+
+    const [gasFee, gasFeesymbol] = useFormatCoin(migrateData?.gasBudget, IOTA_TYPE_ARG);
+    const [
+        totaltotalStorageDepositReturnAmountFormatted,
+        totaltotalStorageDepositReturnAmountSymbol,
+    ] = useFormatCoin(totalNonOwnedStorageDepositReturnAmount.toString(), IOTA_TYPE_ARG);
+
+    async function handleMigrate(): Promise<void> {
+        if (!migrateData) return;
+        signAndExecuteTransaction(
+            {
+                transaction: migrateData.transaction,
+            },
+            {
+                onSuccess: (tx) => {
+                    onSuccess(tx.digest);
```

Comment:
After this, we could show the tx details dialog. We can do it in another issue

Reply:
Thats another issue to speed up this PR https://github.com/iotaledger/iota/issues/4508